src/cpp/rtps/builtin/discovery/participant/PDPServer.cpp

```diff
@@ -636,9 +636,6 @@ void PDPServer::announceParticipantState(
                     metatraffic_locators.add_multicast_locator(locator);
                 }
 
-                // If the DATA is already in the writer's history, then remove it, but do not release the change.
-                remove_change_from_history_nts(mp_PDPWriterHistory, change, false);
-
```

Comment:
`CacheChange_t` is a new one. It will never be in the writer's history. It is unnecessary code. I think it code not removed from a previous implementation.